packages/database-typeorm/__tests__/typeorm-sessions.ts

```diff
@@ -0,0 +1,280 @@
+      expect(ret).toBeTruthy();
+      expect(ret!.id).toBeTruthy();
+      expect(ret!.userId).toEqual(session.userId);
+      expect(ret!.ip).toEqual(session.ip);
+      expect(ret!.userAgent).toEqual(session.userAgent);
+      expect(ret!.valid).toEqual(true);
+      expect(ret!.token).toEqual(token);
+      expect(ret!.createdAt).toEqual(new Date(ret!.createdAt));
+      expect(ret!.updatedAt).toBeTruthy();
+      expect((ret as any).extra).toEqual({ impersonatorUserId });
+    });
+  });
+
+  describe('findSessionById', () => {
+    it('should return null for not found session', async () => {
+      const accountsTypeorm = new AccountsTypeorm({ connection });
+      const ret = await accountsTypeorm.findSessionById('589871d1c9393d445745a57c');
+      expect(ret).not.toBeTruthy();
+    });
+
+    it('should find session', async () => {
+      const accountsTypeorm = new AccountsTypeorm({ connection });
+      const sessionId = await accountsTypeorm.createSession(session.userId, 'token');
+      const ret = await accountsTypeorm.findSessionById(sessionId);
+      expect(ret).toBeTruthy();
+    });
+  });
+
+  describe('findSessionByToken', () => {
+    it('should return null for not found session', async () => {
+      const accountsTypeorm = new AccountsTypeorm({ connection });
+      const ret = await accountsTypeorm.findSessionByToken('589871d1-c939-3d44-5745-a57c1111');
+      expect(ret).not.toBeTruthy();
+    });
+
+    it('should find session', async () => {
+      const accountsTypeorm = new AccountsTypeorm({ connection });
+      const token = generateRandomToken();
+      await accountsTypeorm.createSession(session.userId, token, {
+        ip: session.ip,
+        userAgent: session.userAgent,
+      });
+      const ret = await accountsTypeorm.findSessionByToken(token);
+      expect(ret).toBeTruthy();
+    });
+  });
+
+  describe('updateSession', () => {
+    it('should update session', async () => {
+      const accountsTypeorm = new AccountsTypeorm({ connection });
+      const token = generateRandomToken();
+      const sessionId = await accountsTypeorm.createSession(session.userId, token, {
+        ip: session.ip,
+        userAgent: session.userAgent,
+      });
+      await accountsTypeorm.updateSession(sessionId, {
+        ip: 'new ip',
+        userAgent: 'new user agent',
+      });
+      const ret = await accountsTypeorm.findSessionById(sessionId);
+      expect(ret!.userId).toEqual(session.userId);
+      expect(ret!.ip).toEqual('new ip');
+      expect(ret!.userAgent).toEqual('new user agent');
+      expect(ret!.valid).toEqual(true);
+      expect(ret!.token).toEqual(token);
+      expect(ret!.createdAt).toBeTruthy();
+      expect(ret!.updatedAt).toBeTruthy();
+      expect(ret!.createdAt).not.toEqual(ret!.updatedAt);
+    });
+
+    // TODO: updateSession does not accept a token like mongodb
+    // it('should update session with token', async () => {
```

Comment:
Should this be done before merging?

Reply:
No, same as comment above, I wanted this PR to be only tests without any code changes. Will create separate PRs for other fixes